test/python/circuit/test_instructions.py

```diff
@@ -583,30 +583,30 @@ def dummy_requester(specifier):
         with self.subTest("classical register"):
             instruction = HGate()
             with self.assertWarnsRegex(DeprecationWarning, deprecated_regex):
-                instructions = InstructionSet(registers)
+                instructions = InstructionSet(circuit_cregs=registers)
```

Comment:
I'm ok to change the asserted regex, but if applying the decorator means that the warning isn't triggered by the calling convention previously used in the tests, then we can't do it.  Doing this defeats the purpose of the test: that the standard calling convention issues a deprecation warning.